docs/enhancements/image-service-cloud-authentication.md

```diff
@@ -0,0 +1,164 @@
+* Define an authentication strategy suitable for pre-signed URLs in the cloud. This includes a
+token with a longer expiration (several hours) and a reasonable size (10s to 100s of characters)
+which will be included directly in a download URL
+
+### Non-Goals
+
+* Tie an image download URL to a Red Hat account or organization
+* Maintain public/private key pairs to generate and issue tokens
+* Add additional external identity management components
+
+## Proposal
+
+The new authentication mechanism will be a simple secret and expiration date
+both of which are tied to an infraEnv. This secret will be included as a
+parameter in the image download URL and assisted service authentication will
+pass if the secret in the URL matches the one currently in the infraEnv record
+and the request is processed before the expiration time. A new secret will be
+generated any time a user updates the infraEnv in a way that would change the
+image content. Additionally a new endpoint will be required to force a new
+secret to be created in the event that the current one has expired and no new
+changes to the infraEnv are necessary.
+
+The secret will be a random URL-safe string. In particular it will not be a JWT
+and it will not attempt to encode any information about a user, resources, or its
+own expiration (as a JWT typically does). This approach was chosen to simplify the
+proposal and to eliminate the extra effort and risk associated with managing
+encryption keys for a publicly accessible application. This approach also is
+no less secure than the previous solution where a pre-signed S3 URL could be
+downloaded by anyone with access to the URL itself.
+
+### User Stories
+
+#### Story 1
+
+As a user of assisted service via console.redhat.com, I want a simple download
+URL to access the discovery ISO. This URL should be usable with standard download
+utilities (i.e. `curl`, `wget`) without the need for request headers or additional tools
+like the `ocm` CLI.
+
+#### Story 2
+
+As a user of assisted service via console.redhat.com, I want my sensitive credentials
+and infrastructure information contained within the discovery ISO to be kept secure.
+
+#### Story 3
+
+As a user of assisted service via console.redhat.com, I want the discover ISO download
+URL to be usable by anyone I choose it give it to without disclosing my personal Red Hat
+account credentials or API tokens.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+Assisted service will implement a new security definition to serve this purpose.
+The existing definitions already have well-defined use-cases and scopes that do not
+fully match the requirements of this proposal. Specifically the existing urlAuth security
+definition applies to more endpoints than we would want to expose using just a secret.
+
+This security definition will apply only to the following endpoints:
+
+* `/v2/infra-envs/{infra_env_id}/downloads/files`
+* `/v2/infra-envs/{infra_env_id}/downloads/minimal-initrd`
```

Comment:
Other than the discovery ISO, we have 2 other APIs that require presigned URLS:
GET /v2/clusters/{cluster_id}/downloads/files-presigned
GET /v2/clusters/{cluster_id}/downloads/credentials-presigned (new in v2, similar to files but we moved the kubeconfig and password here because they require stricter permissions)

How would those work?

Reply:
The goal here is not to fully remove all s3 presigned urls. So I don't think those need to change.

We only need to handle authentication for the image download because when we start using the image service it won't be stored in s3 anymore.

Reply:
True